frappe/public/js/frappe/router_history.js

```diff
@@ -5,21 +5,21 @@ const save_routes = frappe.utils.debounce(() => {
 	if (frappe.session.user === 'Guest') return;
 	const routes = frappe.route_history_queue;
 	frappe.route_history_queue = [];
-	
-	frappe.xcall('frappe.deferred_insert.deferred_insert', {
-		'doctype': 'Route History',
-		'records': routes
+
+	if (!routes.length) return;
```

Comment:
This could be moved to line 7. No need to set value to a new array if current one is already empty.